jupyter-book/concluding_remarks.md

```diff
@@ -0,0 +1,261 @@
+- Scikit-learn models are built from a data matrix, of a given number of
+  features for each observation
+
+- Transformations of the data are often necessary
+  - Typically, encoding of the categorical variables
+  - They must only use information available at train time
+  - For this, use the scikit-learn Pipeline object
+
+### 2. Adapting model complexity to the data
+
+- Models seek to minimize the error on the test set
+  - Minimizing error on the train set does no suffice
+  - But too large train error can detect underfit: models too simple for
+    the data
+
+- Models come with multiple hyper-parameters
+  - They can control model complexity
+  - Selecting hyper-parameters is important
+  - In scikit-learn this is done with objects such as GridSearchCV,
+    RandomSearchCV...
+
+### 3. Specific models
+
+- Understanding the models
+  - Helps knowing when they are suited to the data 
+  - Gives intuitions on how to debug them
+
+- Linear models
+  - build predictions by combining the values of features
+  - Particularly useful for data with many features or few observations
+  - Can benefit from non-linear feature engineering
+
+- Tree-based models:
+  - Build predictions by combining a series of binary choices (such as
+    thresholds on the values of the various attributes)
+  - Particularly suited for tabular data, where columns are quantities
+    of different nature, or have missing values
+  - HistGradientBoostingRegressor and Classifier are goto methods that
+    you are strongly advised to check out
+
+
+## Going further with machine learning
+
+Let us give a few pointers on going further with machine learning.
+
+### Learning more about scikit-learn
+
+- The [scikit-learn doc](http://scikit-learn.org)
+    - The documentation is rich, didactic, continuously improving
+- These docs comprise
+  - An user guide: Gives the intuition behind every machine-learning
+    method, and how it can be useful
+  - API docs: Every function, every parameter is explained
+  - Examples: each example tries to demonstrate the good use of the software
+- Where to ask questions:
+    - [Stackoverflow](https://stackoverflow.com/)
+    - [GitHub discussions for scikit-learn](https://github.com/scikit-learn/scikit-learn/discussions)
+
+### We are an open-source community
+
+- Free, open, driven by a community, trying to be inclusive
+- You can contribute
+    - Build a community: helping each other, helping training, communication,
+      advocacy
+    - Curate information: our developers have information overflow
+    - Contributing code is technical
+	- Learn software engineering:
+	- Learn git, github (https://lab.github.com/)
+
+### Topics we have not covered
+
+- Unsupervised learning
+    - Finding order and structure in the data, for instance to group samples, or to transform features
+    - Particularly useful because it does not need labels
+    - But given labels, supervised learning not unsupervised learning, is more likely to recover the link
```

Comment:
I think this line is not clear enough, which link does it refer to?

Reply:
OK, I reformulated.